### PR TITLE
docs(amp-live-list): add documentation for amp-live-list

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
cc @rudygalfi @jmadler

**Comment by erwinmombay** [comment]
@cramforce PTAL

File: extensions/amp-live-list/amp-live-list.md [hunk 1]
```diff
@@ -1 +1,353 @@
...
+children of the `items` reference point are required to have an `id` and a
+`data-sort-time` attribute.
+
+### pagination
+
+The `pagination` reference point is where any sort of pagination markup is located.
+We recommend having a small subtree underneath this reference point as the
+`amp-live-list` component will be doing an inline replace of the DOM received from
+the server in case the page count had increased. We don't do any special
+diffing and just outright do a replace.
```
Comment (cramforce) [inline extensions/amp-live-list/amp-live-list.md]: "just outright replace the contents"

Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

**Comment by erwinmombay** [comment]
cc @ericlindley-g for when he is back

**Comment by erwinmombay** [comment]
@rudygalfi fyi, PR has been reassigned to you

File: extensions/amp-live-list/amp-live-list.md [hunk 2]
```diff
@@ -1 +1,355 @@
...
+component. `update` and `items` are mandatory while `pagination` is optional.
+See "Reference Points" section below for further details.
+
+Example:
+
+```html
+<amp-live-list id="my-live-list" data-poll-interval="15000" data-max-items-per-page="20">
+  <button update on="tap:my-live-list.update">You have updates!</button>
+  <div items></div>
+  <!-- pagination is options -->
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "is optional"?

File: extensions/amp-live-list/amp-live-list.md [hunk 3]
```diff
@@ -1 +1,355 @@
...
+`items` reference point.
+
+This means that publishers of content do not need to set up a JSON endpoint or
+push mechanism for this component to work. New content just needs to be
+published to the same URL, with a valid `amp-live-list` markup, and the user
+will have that content pulled into their client instance during the next poll
+(poll intervals are configurable in the component, and are valid above the
+minimum of 15 seconds).
+
+When multiple `amp-live-list` components are on a page, we choose the lowest
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "smallest polling interval"

File: extensions/amp-live-list/amp-live-list.md [hunk 4]
```diff
@@ -1 +1,355 @@
...
+
+## Reference Points
+
+### update
+
+The `update` reference point is hidden by default (through an `.amp-hidden`
+class which can be overriden) and is shown when new items are received from
+the server. You may style this reference point as a `fixed` position
+item if you want a floating button on the page like on social media websites, to
+draw the readers attention to take action. Currently the `update` reference
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "reader's"

File: extensions/amp-live-list/amp-live-list.md [hunk 5]
```diff
@@ -1 +1,355 @@
...
+newly discovered `id`'s from children of the `items` reference point will turn
+the `update` reference point visible. A user action is required for these items
+to be inserted into the client's live DOM. We scroll the top of the
+`amp-live-list` component into view when the reader clicks on the update action.
+
+If a `data-update-time` attribute is present and its value is a number higher
+than the original `data-sort-time` on the attribute, the item will be updated in
+place through a `replaceChild` operation. If a `data-tombstone` attribute is
+present the element's subtree will be emptied out and the item is hidden through
+css (see amp-live-list.css).
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: linkify amp-live-list.css?

File: extensions/amp-live-list/amp-live-list.md [hunk 6]
```diff
@@ -1 +1,355 @@
...
+
+If a replace or tombstone operation is found but no insert (no new items) is
+found on a poll request, the replace and the tombstone operation will occur
+without the need of for a reader's action. The component and the AMP runtime
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "without the need for a reader's action"

File: extensions/amp-live-list/amp-live-list.md [hunk 7]
```diff
@@ -1 +1,355 @@
...
+have components in its subtree that cause a resize (amp-twitter, amp-iframe,
+etc.) after the fact, we discourage having a lot of replace operations through
+the `data-update-time` attribute.
+
+## How Pagination works
+
+As AMP's focus is on performance we recommend the publisher keep a low number of
+items on a single page by limiting the number of children the `items` reference
+point has, as well as setting a good `data-max-items-per-page` value. The client
+nor the cdn knows nothing about the full number of items so it is up to the
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "nor any cache that may be used"

File: extensions/amp-live-list/amp-live-list.md [hunk 8]
```diff
@@ -1 +1,355 @@
...
+will try and maintain the viewport's scroll position but since a replace could
+have components in its subtree that cause a resize (amp-twitter, amp-iframe,
+etc.) after the fact, we discourage having a lot of replace operations through
+the `data-update-time` attribute.
+
+## How Pagination works
+
+As AMP's focus is on performance we recommend the publisher keep a low number of
+items on a single page by limiting the number of children the `items` reference
+point has, as well as setting a good `data-max-items-per-page` value. The client
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "Neither the client nor..."

File: extensions/amp-live-list/amp-live-list.md [hunk 9]
```diff
@@ -1 +1,355 @@
...
+### Attributes on `amp-live-list`
+
+**id** (Required)
+
+To uniquely identify an amp-live-list (since multiple are allowed on a single
+page).
+
+**data-poll-interval** (Required)
+
+Time (in milliseconds) behind checks for new content (15000 ms minimum is
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: "Time ... interval between checks ..."

File: extensions/amp-live-list/amp-live-list.md [hunk 10]
```diff
@@ -1 +1,355 @@
...
+    </div>
+  </div>
+</amp-live-list>
+
+<amp-live-list id="live-list-2" data-poll-interval="20000" data-max-items-per-page"10">
+  <div update class="slide" on="tap:live-list-2.update">
+    new updates on live list 2
+  </div>
+  <div items>
+    <div id="live-list-2-item-2" data-sort-time="1464281932878">world</div>
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: Just checking: These are supposed to have the same sort time?

Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: yeah, that's perfectly fine. i can change it to make a distinction

**Comment by rudygalfi** [comment]
LGTM on .md file review with some minor suggestions

File: extensions/amp-live-list/amp-live-list.md (hunk 2 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 3 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 4 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 5 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 6 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 7 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 8 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: can you check this one again?

"Neither the client nor any cache that may be used, knows anything about the full number of items..."

File: extensions/amp-live-list/amp-live-list.md [hunk 11]
```diff
@@ -1 +1,355 @@
...
+have components in its subtree that cause a resize (amp-twitter, amp-iframe,
+etc.) after the fact, we discourage having a lot of replace operations through
+the `data-update-time` attribute.
+
+## How Pagination works
+
+As AMP's focus is on performance we recommend the publisher keep a low number of
+items on a single page by limiting the number of children the `items` reference
+point has, as well as setting a good `data-max-items-per-page` value. Neither the client
+nor any cache that may be used, knows anything about the full number of items so
```
Comment (rudygalfi) [inline extensions/amp-live-list/amp-live-list.md]: nor any cache that may be used knows anything about the full number of items, so
(strike comma in middle, add one near end)

**Comment by rudygalfi** [comment]
LGTM

File: extensions/amp-live-list/amp-live-list.md (hunk 11 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 10 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

File: extensions/amp-live-list/amp-live-list.md (hunk 9 above)
Comment (erwinmombay) [inline extensions/amp-live-list/amp-live-list.md]: done.

**Pull request merged**